packages/browser-client/src/store.ts

```diff
@@ -0,0 +1,69 @@
+import { createStore, produce } from 'solid-js/store';
+import { client } from './utils/trpc';
+
+type Lobby = {
+  nickname: string;
+  lobbyCode: string;
+  otherPlayers: string[];
+};
+
+type AppState = {
+  // undefined if you have not joined lobby
+  lobby?: Lobby;
+};
+
+const initialState: AppState = {};
+
+const [_state, setStateStore] = createStore(initialState);
+
+export const state = _state;
+
+export const lobbyJoined = (
+  nickname: string,
+  lobbyCode: string,
+  existingPlayers: string[] = []
+) => {
+  setStateStore(
+    produce((s) => {
+      s.lobby = {
+        nickname,
+        lobbyCode,
+        otherPlayers: existingPlayers,
+      };
+    })
+  );
+  subscribeToLobby();
+};
+
+export const lobbyStateUpdated = (lobby: Lobby) =>
+  setStateStore(
+    produce((s) => {
+      s.lobby = lobby;
+    })
+  );
+
+export const getOtherPlayers = (myNickname: string, nicknames: string[]) =>
+  nicknames.filter((nn) => nn !== myNickname);
+
+const subscribeToLobby = () => {
+  if (state.lobby === undefined) {
+    console.warn('Attempting to subscribe to undefined lobby');
+    return;
+  }
+
+  client.onLobbyChanged.subscribe(
```

Comment:
prob not important yet but we should think about holding onto the subscription so we can unsubscribe when this lobby state is no longer needed